server/api/cartItems.js

```diff
@@ -0,0 +1,103 @@
+const router = require("express").Router();
+const {
+  models: { CartItem, Product, User },
+} = require("../db");
+const { requireAdminToken, requireToken } = require("../auth/index");
+
+//GET /api/cartItems/:userId
+router.get("/:userId", requireToken, async (req, res, next) => {
+  try {
+    const user = await User.findByPk(req.params.userId);
+    if (user.password === req.user.password) {
+      const allCartItems = await CartItem.findAll({
+        include: { model: Product },
+        where: { userId: req.params.userId },
+      });
+      res.send(allCartItems);
+    } else {
+      const error = Error("Unauthorized User");
+      error.status = 401;
+      throw error;
+    }
+  } catch (err) {
+    next(err);
+  }
+});
+
+//PUT /api/cartItems/:cartId
+router.put("/:cartId", requireToken, async (req, res, next) => {
+  try {
+    const updatedQuantity = req.body.quantity;
+    const findCartItem = await CartItem.findByPk(req.params.cartId);
+    const user = await User.findByPk(findCartItem.userId);
+    if (user.password === req.user.password) {
+      await findCartItem.update({ quantity: updatedQuantity });
+      res.send(findCartItem);
+    } else {
+      const error = Error("Unauthorized User");
+      error.status = 401;
+      throw error;
+    }
+  } catch (err) {
+    next(err);
+  }
+});
+
+//POST /api/cartItems
+router.post("/", requireToken, async (req, res, next) => {
+  try {
+    const user = await User.findByPk(req.body.userId);
+    if (user.password === req.user.password) {
+      const newCartItem = req.body;
+      const newItem = await CartItem.create(newCartItem);
```

Comment:
I think we're actually good here. This route assumes that the req.body includes a userId property, so that will automatically be attached on the create cart item.

Reply:
although now that I think about it I could probably just attach the userId from the token auth instead of it needing to be sent in the request